test/e2e/framework/framework.go

```diff
@@ -176,8 +178,20 @@ func createNamespace(client kubeclientset.Interface, baseName string) (string, e
 
 func (f *frameworkWrapper) SetUpServiceDNSControllerFixture() {
 	f.framework().SetUpServiceDNSControllerFixture()
+	f.setUpNamespaceControllerFixture(nil)
 }
 
 func (f *frameworkWrapper) SetUpIngressDNSControllerFixture() {
 	f.framework().SetUpIngressDNSControllerFixture()
+	f.setUpNamespaceControllerFixture(nil)
+}
+
+func (f *frameworkWrapper) setUpNamespaceControllerFixture(typeConfig typeconfig.Interface) {
+	if TestContext.LimitedScope || (typeConfig != nil && typeConfig.GetTarget().Kind == util.NamespaceKind) {
```

Comment:
(no action required) consider adding a comment that this check helps prevent infinite recursion.